gnocchi/carbonara.py

```diff
@@ -408,22 +406,17 @@ class SplitKey(object):
     def __init__(self, value, sampling):
         if isinstance(value, SplitKey):
             self.key = value.key
-        elif isinstance(value, pandas.Timestamp):
-            self.key = value.value / 10e8
-        elif isinstance(value, numpy.datetime64):
-            self.key = (
-                (value - UNIX_UNIVERSAL_START64) / numpy.timedelta64(1, 's')
-            )
         else:
-            self.key = float(value)
+            self.key = value
 
         self.sampling = sampling
```

Comment:
is there a reason you are changing sampling to be a numpy.timedelta and not just float representing seconds? or i guess is there a reason why it only accepts timedelta as sampling? is it possible to split that into another patch to possible shrink this patch?